src/neo/SmartContract/ApplicationEngine.cs

```diff
@@ -202,10 +202,15 @@ public ExecutionContext LoadContract(ContractState contract, string method, Call
             return context;
         }
 
-        public ExecutionContext LoadScript(Script script, CallFlags callFlags, int initialPosition = 0)
+        public ExecutionContext LoadScript(Script script, CallFlags callFlags, UInt160 scriptHash = null, int initialPosition = 0)
```

Comment:
If the `scriptHash` parameter is only used by `LoadContract()`, then the `Hash` should be set in `LoadContract()`, and `LoadScript()` should not be modified.

Reply:
It's also used on `VerifyWitnesses`

Reply:
But `VerifyWitnesses` does not necessarily use it.

Reply:
Also it could be very useful for UT, if your hash will be different than our script hash, I think that we will need it.

Reply:
Maybe we can extend the method in UT instead of making changes here?